src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -341,6 +341,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     If Not CheckTupleMemberName(name, i, argumentSyntax.NameColonEquals.Name, diagnostics, uniqueFieldNames) Then
                         hasErrors = True
                     End If
+
+                    '  check type character
+                    Dim typeChar As TypeCharacter = nameSyntax.Identifier.GetTypeCharacter()
+                    If typeChar <> TypeCharacter.None Then
+                        ReportDiagnostic(diagnostics, nameSyntax, ERRID.ERR_TupleLiteralDisallowsTypeChar)
+                        hasErrors = True
```

Comment:
Can the presence of type character be a source of cascading diagnostics within the same expression. If not, we probably shouldn't set hasErrors to true.